pkg/controller/master/virtualmachine/vm_controller.go

```diff
@@ -377,29 +379,70 @@ func (h *VMController) SetHaltIfInsufficientResourceQuota(_ string, vm *kubevirt
 		return vm, nil
 	}
 
-	if err := h.vmrCalculator.CheckIfVMCanStartByResourceQuota(vm); err == nil {
-		return vm, nil
+	err := h.vmrCalculator.CheckIfVMCanStartByResourceQuota(vm)
+	if err == nil {
+		if vm.Status.PrintableStatus == kubevirtv1.VirtualMachineStatusStarting {
+			logrus.Debugf("VM %s in namespace %s is starting", vm.Name, vm.Namespace)
+			h.vms.EnqueueAfter(vm.Namespace, vm.Name, 5*time.Second)
+			return vm, nil
+		}
+		return vm, h.cleanUpInsufficientResourceAnnotation(vm)
 	} else if !rqutils.IsInsufficientResourceError(err) {
 		return nil, err
 	}
 
-	if err := h.stopVM(vm); err != nil {
-		return nil, err
+	return vm, h.stopVM(vm, err.Error())
+}
+
+func (h *VMController) stopVM(vm *kubevirtv1.VirtualMachine, errMsg string) error {
+	return stopVM(h.vmClient, h.recorder, vm, errMsg)
+}
+
+func (h *VMController) cleanUpInsufficientResourceAnnotation(vm *kubevirtv1.VirtualMachine) error {
+	if vm.Annotations == nil {
+		return nil
+	} else if _, ok := vm.Annotations[util.AnnotationInsufficientResourceQuota]; !ok {
+		return nil
 	}
 
-	h.recorder.Eventf(vm, corev1.EventTypeWarning, "InsufficientResourceQuota", "Set runStrategy to %s", kubevirtv1.RunStrategyHalted)
-	return vm, nil
+	if runStrategy, err := vm.RunStrategy(); err != nil {
+		return err
+	} else if runStrategy == kubevirtv1.RunStrategyHalted {
```

Comment:
why can't we dismiss the annotation when the VM is stopped?

Reply:
If dismiss the annotation when the VM is stopped, it will be displayed for a very short time, cause users don't see the error in time.

Reply:
prefer to use `if` instead of `else if` here